### ISSUE TITLE
Missing Designer Properties - Scale Position

### ISSUE THREAD

**Issue by benmusson** [mussonindustrial/embr#166]
The designer schema does not contain suggestions for `options.scales[id].position`.

https://github.com/mussonindustrial/embr/blob/739afb0db7275b8acf94316d6eb9fc7acf38cb12/modules/charts/common/src/main/resources/chart-js.props.json#L2115C1-L2186C11

**Comment by paul-griffith** [comment]
Consider self-hiding some of this stuff behind `visibleWhen` if appropriate (e.g. this example from the pie chart):
```json
          "showPercentSymbol": {
            "type": "boolean",
            "description": "Whether to show the percent symbol next to the percent value",
            "default": true,
            "visibleWhen": {
              "property": "showValueAsPercent",
              "equals": true
            }
          }
```

**Comment by benmusson** [comment]
How does the work if the properties aren't siblings, or if more nuanced logic is required?

For example:
Line charts have points, so the user should be shown the `data.datasets[id].point~` properties if the chart is a line chart.
The chart is a line chart if `data.datasets[id].type == 'line'`, or if `data.datasets[id].type == undefined` and `type == 'line'`.

**Comment by benmusson** [comment]
@paul-griffith 
Any value in exploring using [PropertySuggestionSourceRegistry](https://files.inductiveautomation.com/sdk/javadoc/ignition81/8.1.38-rc1/com/inductiveautomation/perspective/designer/api/PropertySuggestionSourceRegistry.html)?

Also, what is [DynamicSuggestionsValidator](https://files.inductiveautomation.com/sdk/javadoc/ignition81/8.1.38-rc1/com/inductiveautomation/ignition/common/jsonschema/DynamicSuggestionsValidator.html), and is this something I can use in my own schemas?

**Comment by paul-griffith** [comment]
> How does the work if the properties aren't siblings, or if more nuanced logic is required?

I don't think it does; the `visibleWhen` extension is just something we glued on for our own internal purposes, so if the functionality isn't exposed in a first party use case it's probably not there for third parties either 😢. It _might_ work using relative json path references, but no guarantees.

> Any value in exploring using [PropertySuggestionSourceRegistry](https://files.inductiveautomation.com/sdk/javadoc/ignition81/8.1.38-rc1/com/inductiveautomation/perspective/designer/api/PropertySuggestionSourceRegistry.html)?

Maybe? That's more _intended_ for stuff like "the list of named queries" or "the list of reports" to be populated into the property editor from elsewhere on the backend, but I haven't used your module to know if that might be appropriate.

> Also, what is [DynamicSuggestionsValidator](https://files.inductiveautomation.com/sdk/javadoc/ignition81/8.1.38-rc1/com/inductiveautomation/ignition/common/jsonschema/DynamicSuggestionsValidator.html), and is this something I can use in my own schemas?

DynamicSuggestions are for use _within_ a single component's props - something like "give me the list of configured axes in my axes properties schema":
<img width="836" alt="image" src="https://github.com/user-attachments/assets/32c14bb2-0187-4be1-b053-f52474c27588">

**Comment by benmusson** [comment]
Great, this gives a lot to work on.
